python/tank/authentication/resources/build_resources.sh

```diff
@@ -13,41 +13,85 @@ set -e
 
 # The path to output all built .py files to:
 UI_PYTHON_PATH=../ui
-if [ -z "${PYTHON_BASE}" ]; then
-    PYTHON_BASE="/Applications/Shotgun.app/Contents/Resources/Python"
-fi
-
-# Remove any problematic profiles from pngs.
-for f in *.png; do mogrify $f; done
 
 # Helper functions to build UI files
 function build_qt {
     echo " > Building " $2
-
     # compile ui to python
     $1 $2 > $UI_PYTHON_PATH/$3.py
 
-    # replace PySide imports with local imports and remove line containing Created by date
-    sed -i "" -e "s/from PySide import/from .qt_abstraction import/g" -e "/# Created:/d" $UI_PYTHON_PATH/$3.py
+    # replace PySide2 imports with .qt_abstraction and then added code to set
+    # global variables for each new import.
+    sed -i"" -E \
+        -e "/^from PySide2.QtWidgets(\s.*)?$/d; /^\s*$/d" \
+        -e "s/^(from PySide.\.)(\w*)(.*)$/from .qt_abstraction import \2\nfor name, cls in \2.__dict__.items():\n    if isinstance(cls, type): globals()[name] = cls\n/g" \
+        -e "s/from PySide2 import/from .qt_abstraction import/g" \
+        $UI_PYTHON_PATH/$3.py
 }
 
 function build_ui {
-    build_qt "${PYTHON_BASE}/bin/python ${PYTHON_BASE}/bin/pyside-uic --from-imports" "$1.ui" "../ui/$1"
+    build_qt "$1 -g python --from-imports" "$2.ui" "$2"
 }
 
 function build_res {
-	# Include the "-py3" flag so that we add the `b` prefix to strings for
-	# PySide2 / Python3 compatibility.  This means these files will no longer
-	# be compatible with Python 2.5 and below, but the `b` prefix is ignored in
-	# Python 2.6+.
-    build_qt "${PYTHON_BASE}/bin/pyside-rcc -py3" "$1.qrc" "../ui/$1_rc"
+    build_qt "$1 -g python" "$2.qrc" "$2_rc"
 }
 
+while getopts u:r:p: flag
+do
+    case "${flag}" in
+        u) uic=${OPTARG};;
+        r) rcc=${OPTARG};;
+        p) pyenv=${OPTARG};;
+    esac
+done
+
+if [ -z "${pyenv}" ] && [[ -n "${uic}" && -n "${rcc}" ]]; then
+    pyenv=""
+fi
+
+if [ -z "${pyenv}" ] && [[ -z "${uic}" && -z "${rcc}" ]]; then
+    pyenv="Applications/Shotgun.app/Contents/Resources/Python"
```

Comment:
This path is deprecated since SGD version 1.8 at least. Plus we don't ship the pyside bin utils. So I'd say let's remove that and simplify the code.